rust-runtime/aws-smithy-http/src/byte_stream.rs

```diff
@@ -393,7 +417,9 @@ impl ByteStream {
     /// # }
     /// ```
     pub fn into_async_read(self) -> impl tokio::io::AsyncRead {
-        tokio_util::io::StreamReader::new(self)
+        tokio_util::io::StreamReader::new(
+            crate::futures_stream_adapter::FuturesStreamCompatByteStream::new(self),
```

Comment:
ah I see—this is why we need the bridge. Makes sense. Although this _could_ be pub crate for this use case.

Reply:
Right, and this needs to be `pub` as opposed to `pub(crate)` because codegen server is also [using it](https://github.com/awslabs/smithy-rs/pull/2983/files#diff-3cb0946710c8ff96daf68b3347899e049ffc8dd77bb307c3d4fa43c691a4c49bR32-R36).